equational_theories/ManuallyProved/Equation1729/SmallMagma.lean

```diff
@@ -333,6 +356,11 @@ lemma parent_of_e_sq (a:SM) : parent ((e a)^2) = e a := by
   simp only [parent, e, FreeGroup.toWord_of_pow, List.reduceReplicate, List.tail_cons,
     FreeGroup.mk_of_single_true]
 
+lemma parent_of_div {a b:SM} (h:a ≠ b) : parent ((e a)⁻¹ * e b) = e b := by
+  simp only [parent, e, div_eq, List.singleton_append, FreeGroup.toWord_mk, FreeGroup.reduce.cons,
+    Bool.false_eq, Bool.not_eq_eq_eq_not, Bool.not_false, Bool.true_eq, Bool.not_true,
+    FreeGroup.reduce_nil, h, and_true, ↓reduceIte, List.tail_cons, FreeGroup.mk_of_single_true]
```

Comment:
It can be simplified as follows:

```suggestion
lemma parent_of_div {a b:SM} (h:a ≠ b) : parent ((e a)⁻¹ * e b) = e b := by
  simp [parent, e, div_eq, FreeGroup.reduce_nil, h]
```

Tested locally so it's safe to commit directly.